src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveDocument.java

```diff
@@ -36,4 +50,51 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
                 .optional(CaseDocuments::getDocumentsUploaded)
             .done();
     }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDetails<CaseData, State> details,
+                                                                       final CaseDetails<CaseData, State> beforeDetails) {
+
+        final var beforeCaseData = beforeDetails.getData();
+        final var currentCaseData = details.getData();
+
+        List<ListValue<DivorceDocument>> documentsToRemove = new ArrayList<>();
+
+        var beforeApplicant1DocumentsUploaded = beforeCaseData.getDocuments().getApplicant1DocumentsUploaded();
+        var currentApplicant1DocumentsUploaded = currentCaseData.getDocuments().getApplicant1DocumentsUploaded();
+        if (beforeApplicant1DocumentsUploaded != null & currentApplicant1DocumentsUploaded != null) {
+            beforeApplicant1DocumentsUploaded.forEach(document -> {
+                if (!currentApplicant1DocumentsUploaded.contains(document)) {
+                    documentsToRemove.add(document);
+                }
+            });
+        }
+
+        var beforeDocumentsGenerated = beforeCaseData.getDocuments().getDocumentsGenerated();
+        var currentDocumentsGenerated = currentCaseData.getDocuments().getDocumentsGenerated();
+        if (beforeDocumentsGenerated != null & currentDocumentsGenerated != null) {
+            beforeDocumentsGenerated.forEach(document -> {
+                if (!currentDocumentsGenerated.contains(document)) {
+                    documentsToRemove.add(document);
+                }
+            });
+        }
+
+        var beforeDocumentsUploaded = beforeCaseData.getDocuments().getDocumentsUploaded();
+        var currentDocumentsUploaded = currentCaseData.getDocuments().getDocumentsUploaded();
+        if (beforeDocumentsUploaded != null & currentDocumentsUploaded != null) {
```

Comment:
that's three pretty similar blocks of code, perhaps there could be a method that will diff two lists of documents?

Reply:
Implemented in latest push